### ISSUE TITLE
option to allow all update() calls to callback the listener

### ISSUE THREAD

**Issue by krismeister** [tschaub/hashed#32]
It would be a nice feature if every ``update(state) `` called the listener.

Currently, only real history state changes are broadcast to the listener function. This requires the developer to manually merge the last history change listener state with his new state changes anytime he does ``update()``

This option would allow any update() call to behave the same as a back button click.

**Comment by tschaub** [comment]
The reason this doesn't happen currently is to allow things like the following:

```js
var model = new Model({foo: 'bar'});
var update = hashed.register(model.getState(), function(newState) {
  model.setState(newState);
});
model.on('change', function(state) {
  update(state);
});
```

If a call to `update()` always resulted in a call to the `register()` callback, this would result in an infinite loop.

**Comment by krismeister** [comment]
Yeah there would need to be either a check for this internally when the current hash is the same as what the developer passed in, or rely on that the hashchange event won't fire if the window.location.hash is the same.

I've used that pattern pretty frequently with other frameworks, you treat the hash as state machine instead of just a history manager.

I'm just saying it might be a nice feature.
